torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -540,7 +553,7 @@ struct GraphFuser {
     // we can move the consumer up into the producer.
     // but this requires better handling of merging fusion groups so it is not
     // done now
-    bool shouldFuse = isFusable(producer->node()) &&
+    bool shouldFuse = isFusableDevice(producer) && isFusable(producer->node()) &&
```

Comment:
Why can't we just make this part of the `isFusable` predicate? Check this condition for all used outputs, because they will need to get materialized.